src/core/utils.test.ts

```diff
@@ -47,3 +47,43 @@ describe("filterXSSField", () => {
         expect(result).toEqual({ value: 'test', label: 'test' });
     });
 });
+
+describe('validateNumericInput', () => {
+    let mockPreventDefault: jest.Mock;
+  
+    beforeEach(() => {
+        mockPreventDefault = jest.fn();
+    });
+  
+    test('allows numeric keys', () => {
+      const event = { key: '5', preventDefault: mockPreventDefault } as unknown as React.KeyboardEvent<HTMLInputElement>;
+      
+      validateNumericInput(event);
+
+      expect(mockPreventDefault).not.toHaveBeenCalled();
+    });
+  
+    test('allows navigation keys', () => {
+        const event = { key: 'Tab', preventDefault: mockPreventDefault } as unknown as React.KeyboardEvent<HTMLInputElement>;
+        
+        validateNumericInput(event);
+
+        expect(mockPreventDefault).not.toHaveBeenCalled();
+    });
+  
+    test('prevents non-numeric keys', () => {
+        const event = { key: 'a', preventDefault: mockPreventDefault } as unknown as React.KeyboardEvent<HTMLInputElement>;
+        
+        validateNumericInput(event);
+
+        expect(mockPreventDefault).toHaveBeenCalled();
+    });
+  
+    test('prevents special characters except allowed ones', () => {
```

Comment:
```suggestion
    test('prevents invalid special characters', () => {
```
The way it is currently worded suggests that there are multiple asserts (some for invalid characters and some for valid characters).

Reply:
Updated test description